transmission_interface/package.xml

```diff
@@ -17,17 +17,18 @@
 
   <buildtool_depend>catkin</buildtool_depend>
 
-  <depend>tinyxml</depend>
-  <depend>roscpp</depend>
+  <depend>boost</depend>
```

Comment:
build-depend?

Reply:
It's used in `transmission_interface/transmission_interface_loader.h`

Reply:
Yes, but a header-only dependency doesn't need to be propagated as a runtime dependency. I'm merely proposing to
```suggestion
  <build_depend>boost</build_depend>
```

Reply:
Whoops, yes you are correct.

Reply:
Actually, in this case we need a build_export_depend, since the dependency is used in a public header of this package. I'll give another pass through all the dependencies.

`build_depend` on packages needed to compile (headers & shared libraries)
`build_export_depend` on packages whose headers you transitively include in your public headers
`exec_depend` on packages needed to link (shared libraries)

Reply:
I think the `internal` header above could use just build and build_export_depend for this case

Reply:
Agreed. I found a handful of other mistakes in my last pass at updating dependencies though. In particular, a lot of header-only deps should be updated to build & build_export depend. Currently working on it.

Reply:
Dependencies revised again in 1327dad6. I have one main question coming out of this:

Do we need to `<exec_depend>` on header-only packages containing plugin interfaces (ie. `hardware_interface`), or are `<build_depend>` and `<build_export_depend>` sufficient?

I don't know why the `exec_` would be required, but [the pluginlib wiki](http://wiki.ros.org/pluginlib#pluginlib.2BAC8-pluginlib_groovy.Registering_Plugin_with_ROS_Package_System) states that "the [plugin-]providing package must depend directly on the package containing the plugin interface", with the code example containing both `run_` and `build_` dependencies.

I don't know whether the wiki's advice applies to header-only packages or not, so for now I've simply used `<depend>` as a catch-all. I suspect this is unnecessary, but I preferred to err on the side of caution since I'm unsure.

---

I know this PR is getting long, sorry for dragging it out. I've kinda fallen into the trap of tackling too much in a single PR. Unless you've got insight to resolve this here, I will leave it as is and open a follow-up issue.

Reply:
Follow-up in #414